src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -21,21 +19,30 @@ RandomFourierGaussPreproc::RandomFourierGaussPreproc()
 	    ParameterProperties::HYPER | ParameterProperties::CONSTRAIN,
 	    SG_CONSTRAINT(positive<>()));
 	SG_ADD(
-	    &m_log_width, "log_width", "Kernel width in log domain",
-	    ParameterProperties::HYPER | ParameterProperties::GRADIENT);
+	    &m_kernel, "kernel", "Kernel object",
```

Comment:
you should cast this to `Kernel` type here: `(std::shared_ptr<Kernel>*)&m_kernel`. This allows a user to put/get the Kernel. @karlnapf @vigsterkr Question is if then there should be a callback here checking that the kernel can be casted to `ShiftInvariantKernel` or if that check should be done in fit?

Reply:
Hm, so you're suggesting that m_kernel should be of Kernel shared_ptr type? 

I've put the default args in the header file. 

```shared_ptr<Kernel> m_kernel = std:make_shared<GaussianKerne>() ;``` 

The above would do?

Reply:
I mean you have `std::shared_ptr<ShiftInvariantKernel> m_kernel = std::make_shared<GaussianKernel>();` and whether it should be `std::shared_ptr<Kernel> m_kernel = std::make_shared<GaussianKernel>();`

Reply:
As discussed, that won't work. So we'll work on the constrain thing as you suggested

Reply:
imo we can do this is fit, as the method is not even purely virtual and might throw a runtime error

Reply:
either
* cast to Kernel here and add a constraint
* change the member type to Kernel and check with a dynamic cast in "fit"

I think the first option is better

Reply:
#5082  This should solve our problems :)

Reply:
ok let's wait for that then